src/plugins/intel_npu/src/plugin/npuw/just_sync_infer_request.cpp

```diff
@@ -168,6 +169,40 @@ ov::npuw::JustInferRequest::JustInferRequest(const std::shared_ptr<ov::npuw::Com
         m_subrequests_gio.at(sub_idx).global_results[i] = out_idx;
     }
     // }}}
+
+    // Fill weights bank if applicable
+    if (!m_npuw_model->m_weights_bank_name.empty()) {
+        // FIXME: mostly copied from unpack_closure()
+        for (size_t i = 0; i < m_num_submodels; i++) {
+            LOG_INFO("Filling weights bank for Subgraph[" << i << "]...");
+            LOG_BLOCK();
+            auto& comp_model_desc = m_npuw_model->m_compiled_submodels[i];
+
+            if (!comp_model_desc.compiled_model || (comp_model_desc.replaced_by && comp_model_desc.replaced_by != i)) {
+                OPENVINO_THROW("FATAL: weights bank encountered folded submodel!");
+            }
+
+            auto weights_bank = ov::npuw::weights::bank(m_npuw_model->m_weights_bank_name, *comp_model_desc.device_it);
```

Comment:
it seems you work with a _copy_ of the bank here? Is it implicitly shared?

Anyways, let's move the bank to the compiled model's state.

Reply:
Done